app/services/submission/plant_population_finalizer.rb

```diff
@@ -24,11 +24,12 @@ def create_new_plant_lines
     @new_plant_lines = (submission.content.step03.new_plant_lines || []).map do |attrs|
       attrs = attrs.with_indifferent_access
       taxonomy_term = TaxonomyTerm.find_by!(name: attrs.delete(:taxonomy_term))
+      plant_variety = PlantVariety.find_by!(plant_variety_name: attrs.delete(:plant_variety_name))
       attrs = attrs.merge(
         taxonomy_term_id: taxonomy_term.id,
+        plant_variety_id: plant_variety.id,
         date_entered: Date.today,
         data_provenance: submission.content.step04.data_provenance,
```

Comment:
Some people say it is better to add these extra commas, both ways are fine with me.